src/objects/zcl_abapgit_object_msag.clas.abap

```diff
@@ -179,18 +186,70 @@ CLASS ZCL_ABAPGIT_OBJECT_MSAG IMPLEMENTATION.
 
 
   METHOD zif_abapgit_object~delete.
+    DATA: lv_t100a          TYPE t100a,
+          lv_frozen         TYPE flag,
+          lv_message_id     TYPE arbgb,
+          lv_trkey1         TYPE trkey,
+          lv_access_granted TYPE abap_bool.
+
+* parameter SUPPRESS_DIALOG doesnt exist in all versions of FM RS_DELETE_MESSAGE_ID
+* replaced with a copy
+    lv_message_id = ms_item-obj_name.
+    IF ms_item-obj_name EQ space.
+      zcx_abapgit_exception=>raise( 'Error from (copy of) RS_DELETE_MESSAGE_ID' )."blank message id
+    ELSE.
+      SELECT SINGLE * FROM t100a INTO lv_t100a WHERE arbgb = ms_item-obj_name.
+      IF sy-subrc NE 0.
+        zcx_abapgit_exception=>raise( 'Error from (copy of) RS_DELETE_MESSAGE_ID' )."not found
+      ENDIF.
+    ENDIF.
 
-* parameter SUPPRESS_DIALOG doesnt exist in all versions
-    CALL FUNCTION 'RS_DELETE_MESSAGE_ID'
+    CLEAR lv_frozen.
+    CALL FUNCTION 'RS_ACCESS_PERMISSION'
       EXPORTING
-        nachrichtenklasse = ms_item-obj_name
+        authority_check = 'X'
+        global_lock     = 'X'
+        mode            = 'MODIFY'
+        object          = lv_message_id
+        object_class    = 'T100'
+      IMPORTING
+        frozen          = lv_frozen
       EXCEPTIONS
-        not_executed      = 1
-        not_found         = 2
-        no_permission     = 3
-        OTHERS            = 4.
-    IF sy-subrc <> 0.
-      zcx_abapgit_exception=>raise( 'Error from RS_DELETE_MESSAGE_ID' ).
+        OTHERS          = 1.
+
+    IF sy-subrc NE 0 OR lv_frozen NE space.
+      zcx_abapgit_exception=>raise( 'Error from (copy of) RS_DELETE_MESSAGE_ID' )."can't access
+    ELSE.
+      lv_access_granted = abap_true.
+    ENDIF.
+
+
+
+    CALL FUNCTION 'RS_CORR_INSERT'
+      EXPORTING
+        global_lock        = 'X'
+        object             = lv_message_id
+        object_class       = 'MSAG'
+        mode               = 'D'
+      IMPORTING
+        transport_key      = lv_trkey1
+      EXCEPTIONS
+        cancelled          = 01
+        permission_failure = 02.
+    IF sy-subrc NE 0.
+      IF lv_access_granted = abap_true.
+        free_access_permission( lv_message_id ).
+      ENDIF.
+      zcx_abapgit_exception=>raise( 'Error from (copy of) RS_DELETE_MESSAGE_ID' )."can't access
+    ENDIF.
+
+
+    delete_msgid( lv_message_id ).
+    lv_trkey1-sub_type = '*'.
```

Comment:
is this variable in use after this value assignment?

Reply:
Was a global in the original code, we don't use it at all so I got rid of it